cdap-app-fabric/src/main/java/io/cdap/cdap/internal/operations/OperationRunsStore.java

```diff
@@ -0,0 +1,355 @@
+      Consumer<OperationRunDetail<?>> consumer) throws IOException, OperationRunNotFoundException {
+    Range.Bound startBound = Range.Bound.INCLUSIVE;
+    final Range.Bound endBound = Range.Bound.INCLUSIVE;
+    Collection<Field<?>> startFields = new ArrayList<>();
+
+    startFields.add(
+        Fields.stringField(StoreDefinition.OperationRunsStore.NAMESPACE_FIELD,
+            request.getNamespace())
+    );
+    if (request.getFilter().getOperationType() != null) {
+      startFields.add(Fields.stringField(StoreDefinition.OperationRunsStore.TYPE_FIELD,
+          request.getFilter().getOperationType())
+      );
+    }
+    if (request.getFilter().getStatus() != null) {
+      startFields.add(Fields.stringField(StoreDefinition.OperationRunsStore.STATUS_FIELD,
+          request.getFilter().getStatus().toString()));
+    }
+
+    Collection<Field<?>> endFields = startFields;
+
+    if (request.getScanAfter() != null) {
+      startBound = Range.Bound.EXCLUSIVE;
+      startFields = getRangeFields(
+          new OperationRunId(request.getNamespace(), request.getScanAfter()));
+    }
+
+    Range range = Range.create(endFields, endBound, startFields, startBound);
+    StructuredTable table = getOperationRunsTable(context);
+    String lastKey = null;
+
+    try (CloseableIterator<StructuredRow> iterator = table.scan(range, request.getLimit(),
+        StoreDefinition.OperationRunsStore.START_TIME_FIELD, SortOrder.DESC)) {
+      while (iterator.hasNext()) {
+        StructuredRow row = iterator.next();
+        lastKey = row.getString(StoreDefinition.OperationRunsStore.ID_FIELD);
+        OperationRunDetail<?> detail = GSON.fromJson(
+            row.getString(StoreDefinition.OperationRunsStore.DETAILS_FIELD),
+            OperationRunDetail.class
+        );
+        consumer.accept(detail);
+      }
+    }
+
+    return lastKey;
+  }
+
+  private List<Field<?>> getRangeFields(OperationRunId runId)
+      throws IOException, OperationRunNotFoundException {
+    List<Field<?>> fields = new ArrayList<>();
+    fields.add(
+        Fields.stringField(StoreDefinition.OperationRunsStore.NAMESPACE_FIELD,
+            runId.getNamespace()));
+    fields.add(Fields.stringField(StoreDefinition.OperationRunsStore.ID_FIELD, runId.getRun()));
+
+    Optional<StructuredRow> currentRow = getOperationRunInternal(runId);
+    if (!currentRow.isPresent()) {
+      throw new OperationRunNotFoundException(runId.getNamespace(), runId.getRun());
+    }
+    Long startTime = currentRow.get().getLong(StoreDefinition.OperationRunsStore.START_TIME_FIELD);
+
+    fields.add(
+        Fields.longField(StoreDefinition.OperationRunsStore.START_TIME_FIELD, startTime));
+    return fields;
+  }
+
+  private OperationRunDetail<?> getRunDetail(OperationRunId runId)
+      throws IOException, OperationRunNotFoundException {
+    Optional<StructuredRow> row = getOperationRunInternal(runId);
+    if (!row.isPresent()) {
```

Comment:
Avoid using `isPresent` on Optional. In this case, use `.map` and `orElseThrow`.

```
return getOperationRunInternal(runId)
    .map(this::rowToRunDetail)
    .orElseThrow(() -> new OperationRunNotFoundException(...));
```

Reply:
Please also fix other places that use `isPresent` and replace with more expressive way.

Reply:
Updated. There is one case where we throw error if the row is present. As we can't throw checked error from within `ifPresent` I have not made any change